specs/workspace_spec.rb

```diff
@@ -0,0 +1,85 @@
+require_relative "test_helper"
+
+describe "workspace class" do
+  before do
+    VCR.use_cassette("create workspace object") do
+      selected = "USLACKBOT" #id of slackbot
+      @workspace = Workspace.new(selected: selected)
+    end
+  end
+
+  describe "initialize" do
+    it "creates an instance of workspace" do
+      expect(@workspace).must_be_kind_of Workspace
+    end
+  end
+
+  describe "#select_user" do
+    it "returns an instance of User" do
+      expect(@workspace.select_user).must_be_kind_of User
+    end
+
+    it "returned the correct user" do
+      user = @workspace.select_user
+      expect(user.name).must_equal "slackbot"
+    end
+
+    it "returns nil if username or slack id is not found" do
+      VCR.use_cassette("check nil returned") do
+        selected = "chewy" #id of nonexistent slackbot
+        @workspace2 = Workspace.new(selected: selected)
```

Comment:
You've done a great job of identifying and testing error cases throughout this project.